src/analysis/subspaces/determineBinaryMatrix.m

```diff
@@ -0,0 +1,35 @@
+function [Shat, Shatabs, mconnect, nconnect, mconnectin, mconnectout] = determineBinaryMatrix(S)
+% Determine the binary form of the stoichiometric matrix S
+%
+% INPUT:
+%
+%    S:            Stoichiometric matrix of size m x n (m rows, n columns)
+%
+% OUTPUTS:
+%
+%    Shat:         Binary form of the stoichiometric matrix
+%    Shatabs:      Absolute value of Shat
+%    mconnect:     Compound connectivity (connectivity number; sum of the number of non-zero entries in a row)
+%    nconnect:     Reaction participation number (sum of the number of non-zero entries in a column)
+%    mconnectin:   Produced compound connectivity
+%    mconnectout:  Consumed compound connectivity
+%
+% .. Author: - Laurent Heirendt, June 2018
+
+    Shat = S;  % initialize the binary matrix
+    Shat(S > 0) = 1;  % products, produced -> kin
+    Shat(S < 0) = -1;  % reactants, consumed -> kout
+
+    % determine the row connectivity
+    Shatabs = abs(Shat);
+    if nargout > 2
+        mconnect = sort(sum(Shatabs, 2), 'descend');
```

Comment:
sorting looks very odd here, I would expect the outcome to be linked to each respective row, while by sorting, you actually change that order.

Reply:
I need them to be sorted, but I can specify that in the header.

Reply:
But why put that in this function? Couldn't you sort them afterwards?

Reply:
And why are the other outputs not sorted? This is confusing (imo)

Reply:
I think a flag would make sense (but (imo) that flag should then sort everything, becuase currently this is somewhat of an inconsistency in the function.

Reply:
Yes, I will implement such a flag.